### DISCUSSION TITLE
How to align molecules to a particular 2D orientation

### DISCUSSION THREAD

**Discussion by slochower** [rdkit/rdkit#4120] [Q&A] [answered]
Is there a way to align all molecules to a scaffold _drawn in a particular 2D layout_? To be clear, I'm not talking about aligning the core or MCS of all molecules, but rather, aligning all molecules to a particular – arbitrary but hand-drawn – 2D layout. 

Imagine working on a lead series where all the chemists draw molecules with (for example) the phenol group hydroxyl pointing up to the right and the attached cyclohexane ring down and to the left. 

![image](https://user-images.githubusercontent.com/5393995/117518834-2784e080-af56-11eb-8e4e-61bb6a017678.png)

Is there a way to take this particular layout (say, exported from ChemDraw as a MOL file) and align all molecules to this?

I've scoured the documentation, but have come up emtpy-handed.

My initial attempts have been using `AllChem.AlignMol` but I think I run into trouble that the template molecule doesn't have conformers (and I don't think I want to generate them, since that will erase the hand-drawn orientation):

```python
from rdkit import Chem
template = Chem.MolFromMolFile("template.mol")
template
```
This looks good!
![image](https://user-images.githubusercontent.com/5393995/117519238-a9293e00-af57-11eb-86b1-6f7e6f68aaa5.png)

```python
query = Chem.MolFromSmiles("OC1=CC=C(C2CCC(F)C(Cl)C2Br)C=C1")
query
```
![image](https://user-images.githubusercontent.com/5393995/117519341-f9080500-af57-11eb-85b0-5de1554fb3a2.png)

```python
from rdkit.Chem import rdFMCS
mcs = rdFMCS.FindMCS([template, query])
patt = Chem.MolFromSmarts(mcs.smartsString)

query_match = query.GetSubstructMatch(patt)
template_match = template.GetSubstructMatch(patt)

from rdkit.Chem import AllChem
rms = AllChem.AlignMol(query, template, atomMap=list(zip(query_match,template_match)))

```

Which leads to:

```TraceBack

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-50804ebfdcf9> in <module>
      1 from rdkit.Chem import AllChem
----> 2 rms = AllChem.AlignMol(query, template, atomMap=list(zip(query_match,template_match)))

ValueError: Bad Conformer Id
```

**Comment by slochower** [comment, marked as answer]
Whoops, it seems like all I was missing was

```python
AllChem.Compute2DCoords(query)
```

on the `query` (not the `template`) molecule.

I should leave this up for posterity.

**Reply by ErikCVik** [reply]
I think the following code block should work, although in practice this will rely on the coincidence that template molecule is in the orientation that you want. I tried looking into rotating the molecule (via https://sourceforge.net/p/rdkit/mailman/message/36598250/) but the function `rdMolTransforms.TransformConformer(mol)` doesn't seem to work for non-conformers.

```
from rdkit import Chem
template = Chem.MolFromMolFile("template.mol")
query = Chem.MolFromSmiles("OC1=CC=C(C2CCC(F)C(Cl)C2Br)C=C1")
AllChem.Compute2DCoords(template )
AllChem.GenerateDepictionMatching2DStructure(query , template)
```

**Comment by kathputu** [comment]
Hi,

I have a follow-up question. 
I tried your code and it works fine, however, the matching atoms are highlighted in the resulting structure. Is there a way to turn this off?

Thank you!

**Reply by cpayal** [reply]
This blog could be of help : https://greglandrum.github.io/rdkit-blog/posts/2021-08-07-rgd-and-highlighting.html
